File: app/src/main/java/com/neocaptainnemo/cv/model/CommonResponse.java
package com.neocaptainnemo.cv.model;

import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;

import java.util.List;
import java.util.Map;

@IgnoreExtraProperties
public class CommonResponse {

    @PropertyName("common")
    public Common common;

    @PropertyName("resources")
    public Map<String, Map<String, String>> resources;

    @IgnoreExtraProperties
    public static class Common {
        @PropertyName("sections")
        public List<CommonSection> sections;
    }
}

File: core/src/main/java/com/neocaptainnemo/cv/core/model/CommonResponse.kt
package com.neocaptainnemo.cv.core.model

import androidx.annotation.Keep
import com.google.firebase.database.IgnoreExtraProperties
import com.google.firebase.database.PropertyName

@IgnoreExtraProperties
@Keep
class CommonResponse {

    @PropertyName("common")
    var common: Common? = null

    @PropertyName("resources")
    var resources: HashMap<String, Map<String, String>> = hashMapOf()

    @IgnoreExtraProperties
    @Keep
    class Common {
        @PropertyName("sections")
        var sections: ArrayList<CommonSection> = arrayListOf()
    }
}
